demos/device_defender_for_aws/defender_demo.c

```diff
@@ -0,0 +1,649 @@
+
+/**
+ * @brief The length of #THING_NAME.
+ */
+#define THING_NAME_LENGTH                      ( ( uint16_t ) ( sizeof( THING_NAME ) - 1 ) )
+
+
+/**
+ * @brief Number of seconds to wait for the response from AWS IoT Device
+ * Defender service.
+ */
+#define DEFENDER_RESPONSE_WAIT_SECONDS    ( 2 )
+
+
+/**
+ * @brief Status values of the device defender report.
+ */
+typedef enum
+{
+    ReportStatusNotReceived,
+    ReportStatusAccepted,
+    ReportStatusRejected
+} ReportStatus_t;
+/*-----------------------------------------------------------*/
+
+
+/**
+ * @brief The MQTT context used for MQTT operation.
+ */
+static MQTTContext_t xMqttContext;
+
+/**
+ * @brief The network context used for Openssl operation.
+ */
+static NetworkContext_t xNetworkContext;
+
+/**
+ * @brief Static buffer used to hold MQTT messages being sent and received.
+ */
+static uint8_t ucSharedBuffer[ NETWORK_BUFFER_SIZE ];
+
+/**
+ * @brief Static buffer used to hold MQTT messages being sent and received.
+ */
+static MQTTFixedBuffer_t xBuffer =
+{
+    .pBuffer = ucSharedBuffer,
+    .size    = NETWORK_BUFFER_SIZE
+};
+
+/**
+ * @brief Network Stats.
+ */
+static NetworkStats_t networkStats;
+
+/**
+ * @brief Open TCP ports array.
+ */
+static uint16_t openTcpPorts[ OPEN_TCP_PORTS_ARRAY_SIZE ];
+
+/**
+ * @brief Open UDP ports array.
+ */
+static uint16_t openUdpPorts[ OPEN_UDP_PORTS_ARRAY_SIZE ];
+
+/**
+ * @brief Established connections array.
+ */
+static Connection_t establishedConnections[ ESTABLISHED_CONNECTIONS_ARRAY_SIZE ];
+
+/**
+ * @brief All the metrics sent in the device defender report.
+ */
+static ReportMetrics_t deviceMetrics;
+
+/**
+ * @brief Report status.
+ */
+static ReportStatus_t reportStatus;
+
+/**
+ * @brief Buffer for generating the device defender report.
+ */
+static char deviceMetricsJsonReport[ DEVICE_METRICS_REPORT_BUFFER_SIZE ];
+
+/**
+ * @brief Report Id sent in the defender report.
+ */
+static uint32_t reportId = 0;
+/*-----------------------------------------------------------*/
+
+
+
+/**
+ * @brief Callback to receive the incoming publish messages from the MQTT broker.
+ *
+ * @param[in] pPublishInfo Pointer to publish info of the incoming publish.
+ * @param[in] packetIdentifier Packet identifier of the incoming publish.
+ */
+static void publishCallback( MQTTContext_t * pxMqttContext,
+                             MQTTPacketInfo_t * pxPacketInfo,
+                             MQTTDeserializedInfo_t * pxDeserializedInfo );
+
+/**
+ * @brief Collect all the metrics to be sent in the device defender report.
+ *
+ * @return true if all the metrics are successfully collected;
+ * false otherwise.
+ */
+static bool collectDeviceMetrics( void );
+
+/**
+ * @brief Generate the device defender report.
+ *
+ * @param[out] pOutReportLength Length of the device defender report.
+ *
+ * @return true if the report is generated successfully;
+ * false otherwise.
+ */
+static int generateDeviceMetricsReport( uint32_t * pOutReportLength );
```

Comment:
```suggestion
static bool generateDeviceMetricsReport( uint32_t * pOutReportLength );
```

Reply:
Made a commit to update and align the function commit type with demo helper.
e9d96e2